### ISSUE TITLE
[Question] Gui Style

### ISSUE THREAD

**Issue by 4Magma** [BepInEx/BepInEx.ConfigurationManager#23]
How can i change font size, color and background color?
BCM appears in white on lightgrey in small font size.

**Comment by ManlyMarco** [comment]
You have to compile your own build with a custom gui skin, or make a plugin that hooks config manager and change the gui skin from there.

**Comment by 4Magma** [comment]
Isn't there a easier way? 
I use BCM with Valheim, but the small white text on gray is bad to read.
The Pic on the intro page shows BCM in white on black.

**Comment by ManlyMarco** [comment]
At the moment the game's own skin is used, there's no built in way to change this. The only way would be to change the code to use a custom skin like in runtime editor.

**Comment by Baccanno** [comment]
I'm interested in this too, every UI is very small on my QHD panel.
I got the source code, now I'm just wondering how I should override the GUISkin.

**Comment by ManlyMarco** [comment]
You would have to recalculate sizes of at least fonts for all the different gui styles based on resolution. Create a copy of GUI.skin, make your changes, and then set your copy as the skin to use on every ongui call.

**Comment by Baccanno** [comment]
I would also have to do it for all plugins additional UI calls? It does not seem manageable. Maybe going with dnSpy and changing some values ?
Also, any idea of the base police size on Unity ?

**Comment by ManlyMarco** [comment]
Yes, this would have to be done in every plugin separately. Some plugins would work fine with just a copy of the code, but others would require major rewrites if they use GUI instead of GUILayout and autoscaling windows. A custom skin like this could be exposed from some API library to make it easier to add it to plugins, like what RuntimeUnityEditor does with its own custom skin.

**Comment by Faithfinder** [comment]
Yes please?

**Comment by shaojiaxu** [comment]
> You have to compile your own build with a custom gui skin, or make a plugin that hooks config manager and change the gui skin from there. Something like this https://github.com/ManlyMarco/RuntimeUnityEditor/blob/master/RuntimeUnityEditor/UI/InterfaceMaker.cs#L43

The link is 404 now :(

**Comment by ManlyMarco** [comment]
@shaojiaxu https://github.com/ManlyMarco/RuntimeUnityEditor/blob/093c859332fc02dbab87428090b522b0e5ca515b/RuntimeUnityEditor/UI/InterfaceMaker.cs

**Comment by JoewAlabel** [comment]
I found a good solution for this problem:

1- Find the exe file of your game, and RMB click it.
2- Click at > **Properties**
3- Click at **Compatibility** tab.
4- Click the button: **Change high DPI settings**
5- Check (enable) the first option below Program DPI
6- Check (enable) the last option below High DPI scaling override
7- At this drop-down list, select > System

[Screen Shot](https://prnt.sc/6l2aqGW1-hcD)

**Comment by Stroichik** [comment]
I can add, that if i run Valheim with Vulkan from Steam - i get whitewashed GUI from Config Menu. If i run it from exe or without Vulkan - GUI looks normal colored.

**Comment by ManlyMarco** [comment]
@JoewAlabel @Stroichik Can you check if the latest version of RuntimeUnityEditor also has this GUI issue?

**Comment by Stroichik** [comment]
@ManlyMarco If i run with Vulkan - it looks same whitewashed
![image](https://github.com/BepInEx/BepInEx.ConfigurationManager/assets/12185977/9996a99f-0488-4ff5-9cbe-68d34a608ad1)

And this is how it both looks without Vulkan

![image](https://github.com/BepInEx/BepInEx.ConfigurationManager/assets/12185977/5e97da1e-8d94-498c-bc86-01530bbea889)

Its not something very critical, but a fix would be appreciated

**Comment by ManlyMarco** [comment]
The second screenshot is how it should always look like, I have no idea how Vulkan can affect this outside of it being an Unity bug. I guess it could be partially resolved by having a darker version of the skin but there's no way to do it automatically.

**Comment by Stroichik** [comment]
Well, as i said, its not really that critical.

**Comment by y0soro** [comment]
I have wrote the https://github.com/y0soro/Unity.IMGUI.HiDPI.Patcher to adrress the scaling issue.

Also it's possible to apply method described in https://github.com/y0soro/Unity.IMGUI.HiDPI.Patcher#implement-hidpi-scaling-for-your-imgui-application so BepInEx.ConfigurationManager or any other IMGUI application can have native scaling support.

**Comment by zolantris** [comment]
Can we add this feature and then major version it and feature flag it? Like `enableGuiScaling` force it on in 19.x.x but keep it in compatibility mode until 19.x.x with it off?

**Comment by skullbunnygalaxy** [comment]
> I found a good solution for this problem:
> 
> 1- Find the exe file of your game, and RMB click it. 2- Click at > **Properties** 3- Click at **Compatibility** tab. 4- Click the button: **Change high DPI settings** 5- Check (enable) the first option below Program DPI 6- Check (enable) the last option below High DPI scaling override 7- At this drop-down list, select > System
> 
> [Screen Shot](https://prnt.sc/6l2aqGW1-hcD)

For anyone else who has this issue because they are using a handheld device (ie the Claw or a Steamdeck) and connecting it to a tv, I am adding the additional setting of adjusting the dropdown to "When I open this program" instead of the default "I signed in to Windows".

<img width="1269" height="1599" alt="Image" src="https://github.com/user-attachments/assets/f110903f-bb9f-45f7-bf60-f6df94c1111c" />

This workaround does ruin the scaling in other parts of the game when playing Sunhaven, but there is a Sunhaven mod called [UIScales](https://www.nexusmods.com/sunhaven/mods/7) that allows adjusting the in-game UI.
